libazureinit/src/kvp.rs

```diff
@@ -119,25 +121,43 @@ impl Kvp {
                 biased;
 
                 Some(encoded_kvp) = events.recv() => {
+                    if let Err(e) = FileExt::lock_exclusive(&file) {
+                        eprintln!("Failed to lock KVP file: {e}. Dropping KVP message.");
+                        continue;
+                    }
+
                     if let Err(e) = file.write_all(&encoded_kvp) {
                         eprintln!("Failed to write to log file: {e}");
                     }
                     if let Err(e) = file.flush() {
                          eprintln!("Failed to flush the log file: {e}");
                     }
+
+                    if let Err(e) = FileExt::unlock(&file) {
+                        eprintln!("Failed to unlock KVP file: {e}");
+                    }
                 }
 
                 _ = token.cancelled() => {
                     // Shutdown signal received.
                     // close the channel and drain remaining messages.
                     events.close();
                     while let Some(encoded_kvp) = events.recv().await {
+                        if let Err(e) = FileExt::lock_exclusive(&file) {
```

Comment:
given the duplicate pattern here I suspect there should be write_kvps() method that handles draining available kvps while only taking the lock and calling flush once